core/src/main/java/org/apache/iceberg/PartitionStatsHandler.java

```diff
@@ -239,30 +263,30 @@ private static OutputFile newPartitionStatsFile(
   private static PartitionStats recordToPartitionStats(StructLike record) {
     PartitionStats stats =
         new PartitionStats(
-            record.get(PARTITION_FIELD_ID, StructLike.class),
-            record.get(SPEC_ID.fieldId(), Integer.class));
-    stats.set(DATA_RECORD_COUNT.fieldId(), record.get(DATA_RECORD_COUNT.fieldId(), Long.class));
-    stats.set(DATA_FILE_COUNT.fieldId(), record.get(DATA_FILE_COUNT.fieldId(), Integer.class));
+            record.get(PARTITION_POSITION, StructLike.class),
+            record.get(SPEC_ID_POSITION, Integer.class));
+    stats.set(DATA_RECORD_COUNT_POSITION, record.get(DATA_RECORD_COUNT_POSITION, Long.class));
+    stats.set(DATA_FILE_COUNT_POSITION, record.get(DATA_FILE_COUNT_POSITION, Integer.class));
     stats.set(
-        TOTAL_DATA_FILE_SIZE_IN_BYTES.fieldId(),
-        record.get(TOTAL_DATA_FILE_SIZE_IN_BYTES.fieldId(), Long.class));
+        TOTAL_DATA_FILE_SIZE_IN_BYTES_POSITION,
+        record.get(TOTAL_DATA_FILE_SIZE_IN_BYTES_POSITION, Long.class));
     stats.set(
-        POSITION_DELETE_RECORD_COUNT.fieldId(),
-        record.get(POSITION_DELETE_RECORD_COUNT.fieldId(), Long.class));
+        POSITION_DELETE_RECORD_COUNT_POSITION,
+        record.get(POSITION_DELETE_RECORD_COUNT_POSITION, Long.class));
     stats.set(
-        POSITION_DELETE_FILE_COUNT.fieldId(),
-        record.get(POSITION_DELETE_FILE_COUNT.fieldId(), Integer.class));
+        POSITION_DELETE_FILE_COUNT_POSITION,
+        record.get(POSITION_DELETE_FILE_COUNT_POSITION, Integer.class));
     stats.set(
-        EQUALITY_DELETE_RECORD_COUNT.fieldId(),
-        record.get(EQUALITY_DELETE_RECORD_COUNT.fieldId(), Long.class));
+        EQUALITY_DELETE_RECORD_COUNT_POSITION,
+        record.get(EQUALITY_DELETE_RECORD_COUNT_POSITION, Long.class));
     stats.set(
-        EQUALITY_DELETE_FILE_COUNT.fieldId(),
-        record.get(EQUALITY_DELETE_FILE_COUNT.fieldId(), Integer.class));
-    stats.set(TOTAL_RECORD_COUNT.fieldId(), record.get(TOTAL_RECORD_COUNT.fieldId(), Long.class));
-    stats.set(LAST_UPDATED_AT.fieldId(), record.get(LAST_UPDATED_AT.fieldId(), Long.class));
+        EQUALITY_DELETE_FILE_COUNT_POSITION,
+        record.get(EQUALITY_DELETE_FILE_COUNT_POSITION, Integer.class));
+    stats.set(TOTAL_RECORD_COUNT_POSITION, record.get(TOTAL_RECORD_COUNT_POSITION, Long.class));
+    stats.set(LAST_UPDATED_AT_POSITION, record.get(LAST_UPDATED_AT_POSITION, Long.class));
     stats.set(
-        LAST_UPDATED_SNAPSHOT_ID.fieldId(),
-        record.get(LAST_UPDATED_SNAPSHOT_ID.fieldId(), Long.class));
+        LAST_UPDATED_SNAPSHOT_ID_POSITION,
+        record.get(LAST_UPDATED_SNAPSHOT_ID_POSITION, Long.class));
```

Comment:
Could we just do this?
```
    PartitionStats stats =
        new PartitionStats(
            record.get(PARTITION_POSITION, StructLike.class),
            record.get(SPEC_ID_POSITION, Integer.class));
    for(int i = 0; i<record.size(); ++i) {
      stats.set(i, record.get(i, Object.class));
    }
```

Reply:
`i` should start from `2`?

Reply:
yes, for sure. But I got what he mean and updated locally.

Reply:
`SPEC_ID_POSITION + 1`